common/cache/lru.go

```diff
@@ -300,20 +306,22 @@ func (c *lru) putInternal(key interface{}, value interface{}, allowUpdate bool)
 		entry.createTime = c.clock.Now().UTC()
 	}
 
-	if len(c.byKey) >= c.maxSize {
+	if c.currSize >= c.maxSize {
 		c.evictOnceInternal()
 	}
```

Comment:
`if` should be changed to `for` loop?
e.g. what if a newly added item force evict 2 old items?

Reply:
i guess at the very beginning of putting an item into cache
1. check if item size >= cache max size (or max allowed item size) -> if so error out earlier
2. if after adding the item, if cache size exceed limit, loop evicting old items until cache size <= threshold